### PR TITLE
beacon-chain, config: get consensus values from beacon config

### PR DESCRIPTION
**Pull request by gballet**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

This allows prysm to get some of its consensus parameters from the config file, instead of getting them from a hardcoded file. This is necessary to support e.g. the gnosis chain, that has a different slot time.

Another approach I considered was to store the hardcoded values for the gnosis chain inside a file in `config/fieldparams`, protected by a tag. I think this PR is the better approach, however, as adding other chains would require adding a file each time, when the config file can be tweaked instead.

### CODE REVIEW

**Review by terencechain — APPROVED** [review]
Ty!

**Review by nisdas — CHANGES_REQUESTED** [review]
I will need to think on this more, but the main reason it is a hardcoded value rather than something that can be configured at runtime is because presets are meant to be initialized at compile time. For context, you can take a look at this PR:
https://github.com/prysmaticlabs/prysm/pull/10007

I understand, that this won't work with gnosis chain currently which requires it being changed at runtime.

**Comment by gballet** [comment]
That makes sense, I'll work on the gnosis preset then.

File: config/params/config.go [hunk 1]
```diff
@@ -239,3 +239,15 @@ func configForkNames(b *BeaconChainConfig) map[[fieldparams.VersionLength]byte]s
 	fvn[bytesutil.ToBytes4(b.BellatrixForkVersion)] = "bellatrix"
 	return fvn
 }
+
+func (b *BeaconChainConfig) Eth1DataVotesLength() uint64 {
```
Comment (nisdas) [inline config/params/config.go:243]: Lets add a comment for each new method

**Review by nisdas — COMMENTED** [review]
@gballet So we talked internally and while the change isn't great, we would be okay with this. Given that we are the only client that has this strict requirement, it seems that it has limited benefit to forcing this to be a compile time constant

**Comment by gballet** [comment]
@nisdas thanks for your feedback, I have added the required comments. Let me know if you would like other changes.

**Review by nisdas — APPROVED** [review]

**Pull request merged**